src/storage.ts

```diff
@@ -123,10 +127,44 @@ export class LlamaStorage {
   }
 
   async list(): Promise<FileRecord[]> {
-    return this.db.getAll('metadata');
+    const localMeta = this.db.getAll('metadata');
+    let firebaseList: firebase.ListResult | null = null;
+    if (firebaseStorage) {
+      firebaseList = await firebase.listAll(firebase.ref(firebaseStorage, firebaseDir));
+    }
+
+    const results = await localMeta;
+    if (firebaseList === null) {
+      return results;
+    }
+
+    // TODO: this is potentially high latency - can we load asynchronously?
+    for (const res of firebaseList.items) {
+      const meta = await this.getRemoteFirebaseMetadata(res);
+      if (meta) {
+        results.push(meta);
+      }
+    }
+
+    const m: { [key: string]: boolean } = {};
+    return results.filter((v: FileRecord) => {
+      let id = v.id;
+      if (id.indexOf(remoteIdPrefix) === 0) {
```

Comment:
nit: here and elsewhere, `id.startsWith(remoteIdPrefix)` ([mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith)) which reads a lot more intuitively.

Reply:
Done.